src/keri/mod.rs

```diff
@@ -216,4 +423,40 @@ impl<'d, K: KeyManager> Keri<'d, K> {
     pub fn get_state_for_seal(&self, seal: &EventSeal) -> Result<Option<IdentifierState>, Error> {
         self.processor.compute_state_at_sn(&seal.prefix, seal.sn)
     }
+
+    fn generate_ntr(&self, message: EventMessage) -> Result<SignedNontransferableReceipt, Error> {
+        let signature = self.key_manager.borrow().sign(&message.serialize()?)?;
+        let bp = BasicPrefix::new(Basic::Ed25519, self.key_manager.borrow().public_key());
+        let ssp = SelfSigningPrefix::new(SelfSigning::Ed25519Sha512, signature);
+        let ntr = SignedNontransferableReceipt::new(&message, vec!((bp, ssp)));
```

Comment:
`ntr` after serialization is receipted message with witness signatures attached, for example: 

```{"v":"KERI10JSON0000ed_","i":"D90ITuRKvy5HsisC-ilKVJ6yZ0kcA-4c8iQX7B4Ghoy8","s":"0","t":"icp","kt":"1","k":["D90ITuRKvy5HsisC-ilKVJ6yZ0kcA-4c8iQX7B4Ghoy8"],"n":"ECrz6oVftsIb2A_E4wBgcrmdZar775ExviUhKsHJmJAk","bt":"0","b":[],"c":[],"a":[]}-AB0BfOVKw0lNl1T9P00p9E7rEoxFj4eOS2cFUbfP7Riv_4PWYI_mc9geGM7MfwaTAQmklNk3I7IeQZrO8nFYZL9zAA```

Shouldn't it be receipt message with witness signatures? For example something like this:

`{"v":"KERI10JSON000091_","i":"EpU9D_puIW_QhgOf3WKUy-gXQnXeTQcJCO_Igcxi1YBg","s":"0","t":"rct","d":"EIt0xQQf-o-9E1B9VTDHiicQzVWk1CptvnewcnuhSd0M"}-BADAACZrTPLvG7sNaxtV8ZGdIHABFHCZ9FlnG6b4J6a9GcyzJIJOjuGNphW2zyC_WWU6CGMG7V52UeJxPqLpaYdP7CgAB8npsG58rX1ex73gaGe-jvRnw58RQGsDLzoSXaGn-kHRRNu6Kb44zXDtMnx-_8CjnHqskvDbz6pbEbed3JTOnCQACM4bMcLjcDtD0fmLOGDx2oxBloc2FujbyllA7GuPLm-RQbyPPQr70_Y7DXzlWgs8gaYotUATeR-dj1ru9qFwADA`

Reply:
Not sure how you've got to `ntr` serialization, but this PR does not include any changes to that part.
Should we file separate issue to get that aligned?

Reply:
`generate_nrt` function was added. I noticed that receipt here is constructed with receipted message, not receipt message. `ntr` serialization is probably ok ;)

Reply:
so what's the actual change needed here? =)

Reply:
We should construct receipt event message 

```
let rcp = Event {
            prefix: message.event.prefix.clone(),
            sn: message.event.sn,
            event_data: EventData::Rct(Receipt {
                receipted_event_digest: SelfAddressing::Blake3_256.derive(&message.serialize()?),
            }),
        }
        .to_message(SerializationFormats::JSON)?;
```
and use it to create `SignedNontransferableReceipt`

```suggestion
        let ntr = SignedNontransferableReceipt::new(&rcp, vec!((bp, ssp)));
```